download-basket/src/main/java/downloadbasket/actions/GetFeatureForCropping.java

```diff
@@ -67,6 +72,9 @@ public void handleAction(final ActionParameters params) throws ActionException {
 			try {
 
 				HttpURLConnection con = IOHelper.getConnection(wmsUrl);
+				if(hasAuthorization()){
```

Comment:
The same username/password combination is used for all layers. Is this intentional?

Reply:
Yes, it is intentional. Example case: a GeoServer workspace is restricted via WFS, so all the layers inside the workspace can be accessed with the same authentication credentials.

In ext-properties, the workspace is defined by its namespace: oskari.wfs.cropping.namespace = cropping_data

Reply:
You should use oskariLayer.getUser/passwd instead as you are doing a GFI response.